src/rust/ide/view/graph-editor/src/component/visualization/foreign/java_script/visualization.js

```diff
@@ -1,10 +1,51 @@
 export class Visualization {
     constructor(api) {
+        // These go before `api` assignment so the `undefined` is not emitted to IDE.
+        // First we will give deriving type a chance to overwrite them, then IDE will
+        // invoke `emitPreprocessorChange()` on this.
+        this.__preprocessorCode__ = undefined
+        this.__preprocessorModule__ = undefined
```

Comment:
This is antipattern in JS. You sshould never use explicitly "undefined" - it means that a variable doesn't exist. Use `null` instead.